File: gang/week2/직사각형 넓이 구하기.java
class Solution {
    public int solution(int[][] dots) {
        int answer = 0;
        int side1 = sideLength(dots[0], dots[1]);   //첫번째와 두번째 배열 비교
        int side2 = sideLength(dots[1], dots[2]);   //두번째와 세번째 배열 비교
        answer = side1 * side2; //나온 변의 길이를 곱하면 넓이가 된다.
        return answer;
    }

    public int sideLength(int[] dot1, int[] dot2) { //변의 길이를 구하는 함수
        int side = 0;
        if(dot1[0] == dot2[0]) {    //배열 두개를 비교해서 x의 값이 서로 같으면
            side = Math.abs(dot1[1]-dot2[1]);   //y축 변의 길이 구하기
        }
        else {  //반대일 경우
            side = Math.abs(dot1[0]-dot2[0]);   //x축 변의 길이 구하기
        }
        return side;
    }
}

File: jeonghyeon/2week/직사각형 넓이 구하기.js
function solution(dots) {
    // map() : 주어진 함수를 호출해 새로운 배열 생성
    const x = dots.map(([x, _]) => x); // dots 배열의 요소들의 첫 번째 값 뽑아 x 배열에 저장
    const y = dots.map(([_, y]) => y); // dots 배열의 요소들의 두 번째 값 뽑아 y 배열에 저장
    // 사용하지 않는 값들은 _ 로 표시

    return (Math.max(...x)- Math.min(...x)) * (Math.max(...y)- Math.min(...y)); 
    // 각 x, y좌표들의 최댓값과 최솟값을 구해 가로, 세로 길이 구한 후 곱해서 넓이 구하기
}